mm2src/common/log.rs

```diff
@@ -171,18 +174,11 @@ pub fn short_log_time(ms: u64) -> DelayedFormat<StrftimeItems<'static>> {
 #[macro_export]
 macro_rules! log {
     ($($args: tt)+) => {{
-        use std::fmt::Write;
-
-        // We can optimize this with a stack-allocated SmallVec from https://github.com/arcnmx/stack-rs,
-        // though it doesn't worth the trouble at the moment.
-        let mut buf = String::new();
-        wite! (&mut buf,
-            ($crate::log::short_log_time ($crate::now_ms()))
-            if cfg! (target_arch = "wasm32") {"ʷ "} else {", "}
-            (::gstuff::filename (file!())) ':' (line!()) "] "
-            $($args)+)
-        .unwrap();
-        $crate::log::chunk2log(buf, $crate::log::LogLevel::Info)
+        let time = $crate::log::short_log_time($crate::now_ms());
+        let file = ::gstuff::filename (file!());
```

Comment:
Can we maybe also remove `gstuff` usage here? 🙂

Reply:
It's used in many places. Do you suggest moving it to `common` crate?

Reply:
I think gstuff::filename is much more efficient than [Path::file_name](https://doc.rust-lang.org/stable/std/path/struct.Path.html#method.file_name), because it's based on two operations:
1) Find last `/` or `\` in the string slice
2) Checks if it's needed to return a slice without `.rs` suffix

My benchmarks:
gstuff::filename, 1000 iterations: 349.083µs
Path::file_name, 1000 iterations: 1.693791ms

gstuff::filename, 10000 iterations: 4.311166ms
Path::file_name, 10000 iterations: 19.391583ms

How about moving `gstuff::filename` function to `common.rs`?

Reply:
> How about moving gstuff::filename function to common.rs?

Ok :)

Reply:
Done